Exoft.Gamification.Api.Services/AuthService.cs

```diff
@@ -14,24 +17,46 @@ namespace Exoft.Gamification.Api.Services
     public class AuthService : IAuthService
     {
         private readonly IUserRepository _userRepository;
+        private readonly IAuthRepository _authRepository;
+        private readonly IUnitOfWork _unitOfWork;
         private readonly IJwtSecret _jwtSecret;
 
         public AuthService
         (
             IUserRepository userRepository,
+            IAuthRepository authRepository,
+            IUnitOfWork unitOfWork,
             IJwtSecret jwtSecret
         )
         {
             _userRepository = userRepository;
+            _authRepository = authRepository;
+            _unitOfWork = unitOfWork;
             _jwtSecret = jwtSecret;
         }
 
         public async Task<JwtTokenModel> Authenticate(string userName, string password)
         {
             var userEntity = await _userRepository.GetByUserNameAsync(userName);
 
-            //TODO: provide password hash
-            if (userEntity == null || password != userEntity.Password)
+            // refresh token
+            var refreshTokenFromDB = await _authRepository.GetByUserIdAsync(userEntity.Id);
+            if (refreshTokenFromDB != null)
+            {
+                _authRepository.Delete(refreshTokenFromDB);
+                await _unitOfWork.SaveChangesAsync();
```

Comment:
Why don't just clear all users tokens?
Also I see here SaveChanges called twice, why it happens?